auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue

```diff
@@ -97,60 +158,219 @@
+      } catch (error) {
+        console.error(error)
+      } finally {
+        state.isTableLoading = false
+      }
+    })
+
+    function mounted () {
+      state.tableDataOptions = DEFAULT_DATA_OPTIONS
+      const reviewSearchFilter = ConfigHelper.getFromSession(props.reviewSessionStorageKey) || ''
+      try {
+        state.reviewParams = JSON.parse(reviewSearchFilter)
+      } catch {
+        // Do nothing, we have defaults for review searchParams.
+      }
+      const hasInfo = hasCachedPageInfo(props.reviewPaginationOptionsKey)
+      if (hasInfo) {
+        state.tableDataOptions = getAndPruneCachedPageInfo(props.reviewPaginationOptionsKey)
+      }
     }
 
-    function view (reviewId: string) {
-      // route to Continuation Authorization Review page
+    watch(() => state.reviewParams, function (value) {
+      state.searchParamsExist = doSearchParametersExist(value)
+      state.tableDataOptions = { ...getAndPruneCachedPageInfo(props.reviewPaginationOptionsKey), page: 1 }
+      setSearchFilterToStorage(JSON.stringify(value))
+      debouncedOrgSearch()
+    }, { deep: true })
+
+    watch(() => state.tableDataOptions, function (val) {
+      debouncedOrgSearch(val?.page, val?.itemsPerPage)
+    }, { deep: true })
+
+    function getButtonLabel (status: string) {
+      const reviewStates = ['AWAITING_REVIEW', 'RESUBMITTED']
+      return reviewStates.includes(status) ? 'Review' : 'View'
+    }
+    async function view (reviewId: string) {
       this.$router.push(`/staff/continuation-review/${reviewId}`)
     }
 
+    function clearSearchParams () {
+      state.reviewParams = {
+        ...state.reviewParams,
+        submissionDate: '',
+        nrNumber: '',
+        identifier: '',
+        completingParty: '',
+        status: ''
+      }
+    }
+
     function getIndexedTag (tag: string, index: number): string {
       return `${tag}-${index}`
     }
 
+    function updateItemsPerPage (options: DataOptions): void {
+      cachePageInfo(options, props.reviewPaginationOptionsKey)
+    }
+
+    const noDataMessage = computed(() => {
+      return t(
+        state.searchParamsExist
+          ? 'searchReviewsNoResult'
+          : 'searchReviewStartMessage'
+      )
+    })
+
+    function setSearchFilterToStorage (val: string): void {
+      ConfigHelper.addToSession(props.reviewSessionStorageKey, val)
+    }
+
+    function doSearchParametersExist (params: ReviewFilterParams): boolean {
+      return params.submissionDate.length > 0 ||
+                params.nrNumber.length > 0 ||
+                params.identifier.length > 0 ||
+                params.completingParty.length > 0 ||
+                params.status.length > 0
+    }
+
+    // Method to display the status
+    function displayStatus (status: string): string {
+      return status
+        .toLowerCase()
+        .replace(/_/g, ' ')
+        .replace(/(^\w{1})|(\s+\w{1})/g, letter => letter.toUpperCase())
```

Comment:
This is giving a [security warning](https://sonarcloud.io/project/security_hotspots?id=bcgov_sbc-auth&pullRequest=2928&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true).

Maybe it would be easier to map "status" using a small lookup object?

Reply:
Thanks, will do. A quick question: the Figma design used "Accepted", but the data fetched from API and in enum ReviewStatus use "Approved". Which should we use?
![image](https://github.com/user-attachments/assets/c63d0bed-124f-4d35-a5a6-316b0f2eea4a)
Figma:
![image](https://github.com/user-attachments/assets/2480cf52-544b-48ba-a09d-af1ef48b72c8)

Reply:
I've gone with "approved" on the review page. Can you double-check with Andy?

Reply:
Confirmed with Andy, use "Approved"

Reply:
Updated to display status using a map

Reply:
I don't know why SonarCloud didn't automatically pick up the fixed code and I don't know how to re-rerun the analysis, so I went in to SonarCloud and told it the code was fixed.